src/utils/daemon.js

```diff
@@ -61,3 +66,48 @@ export default async function createDaemon (opts) {
 
   return ipfsd
 }
+
+async function cleanup (opts) {
+  const option = dialog.showMessageBox({
+    type: 'warning',
+    title: 'IPFS Desktop',
+    message: `We coudln't start because IPFS didn't shutdown correctly last time. We can try to
+solve the issue by running 'ipfs repo fsck'. Would you like to?`,
+    buttons: [
+      'No',
+      'Yes, run "ipfs repo fsck"'
+    ],
+    cancelId: 0
+  })
+
+  if (option === 0) {
+    return false
+  }
+
+  const exec = findExecutable('go', join(__dirname, '..'))
+  spawnSync(exec, ['repo', 'fsck'], {
+    env: {
+      ...process.env,
+      IPFS_PATH: opts.path
+    }
+  })
+
+  return true
+}
+
+export default async function (opts) {
+  try {
+    const ipfsd = await createDaemon(opts)
+    return ipfsd
+  } catch (e) {
+    if (!e.message.includes('ECONNREFUSED')) {
```

Comment:
I don't think looking for just `ECONNREFUSED' gives us enough info. I think we want to do the work of checking if there is an `api` file and no repo lock file as well, otherwise this could be a case of not being able to connect to an api for a daemon that _is_ running. Offering to `fsck` the repo in that case would be unhelpful. @lidel what do you think?

I still think we should run `repo fsck` automatically in the common case wherer there is no lockfile and the api file points to the same location as the API address in the config file.

Reply:
Running `repo fsck` to fix `ECONNREFUSED` makes sense only if:
- `api` is present 
  - AND points at the same multiaddr as `config`  
    - AND (`$IPFS_PATH/repo.lock` OR `$IPFS_PATH/datastore/LOCK`) is present

Reply:
@lidel if is present and points to the **same** address? I thought that in that case it was safe to run it without consent of the user. I was thinking we should ask them only if it was different.

Reply:
yes, sorry, I meant "not the same", fixed my comment :)